pkg/cmd/context.go

```diff
@@ -2,49 +2,143 @@ package cmd
-			level.Info(logger).Log("msg", "context called")
-		},
+		Short: "Manage context configuration.",
+		Long:  "Manage context configuration.",
 	}
 
 	apiCmd := &cobra.Command{
 		Use:   "api",
 		Short: "Add/edit API configuration.",
 		Long:  "Add/edit API configuration.",
-		Run: func(cmd *cobra.Command, args []string) {
-			level.Info(logger).Log("msg", "api called")
+	}
+
+	var addURL, addName string
+	apiAddCmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add API configuration.",
+		Long:  "Add API configuration.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			return conf.AddAPI(logger, addName, addURL)
 		},
 	}
+
+	apiAddCmd.Flags().StringVar(&addURL, "url", "", "The URL for the Observatorium API.")
+	apiAddCmd.Flags().StringVar(&addName, "name", "", "Provide an optional name to easily refer to the Observatorium Instance.")
+
+	err := apiAddCmd.MarkFlagRequired("url")
+	if err != nil {
+		panic(err)
+	}
+
+	var rmName string
+	apiRmCmd := &cobra.Command{
+		Use:   "rm",
+		Short: "Remove API configuration.",
+		Long:  "Remove API configuration. If only one API is saved, that will be removed. If set to current, current will be set to nil.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			return conf.RemoveAPI(logger, rmName)
+		},
+	}
+
+	apiRmCmd.Flags().StringVar(&rmName, "name", "", "The name of the Observatorium API instance to remove. No need to provide if only one API is saved.")
+
 	switchCmd := &cobra.Command{
 		Use:   "switch",
 		Short: "Switch to another context.",
 		Long:  "View/Add/Edit context configuration.",
-		Run: func(cmd *cobra.Command, args []string) {
-			level.Info(logger).Log("msg", "switch called")
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cntxt := strings.Split(args[0], "/")
+			if len(cntxt) != 2 {
+				return fmt.Errorf("invalid context name: use format <api>/<tenant>")
+			}
+
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			return conf.SetCurrentContext(logger, cntxt[0], cntxt[1])
 		},
 	}
+
 	currentCmd := &cobra.Command{
 		Use:   "current",
 		Short: "View current context configuration.",
 		Long:  "View current context configuration.",
-		Run: func(cmd *cobra.Command, args []string) {
-			level.Info(logger).Log("msg", "current called")
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			_, _, err = conf.GetCurrentContext()
+			if err != nil {
+				return err
+			}
+
+			// TODO(saswatamcode): Add flag to display more details. Eg -verbose
+			fmt.Fprintf(os.Stdout, "The current context is: %s/%s\n", conf.Current.API, conf.Current.Tenant)
+			return nil
+		},
+	}
+
+	listCmd := &cobra.Command{
+		Use:   "list",
```

Comment:
I was also wondering whether it would make sense to add `list` command for APIs as well?

Reply:
The list command lists out all possible context in the form of,
```
API1:
    - tenant1
    - tenant2
API2:
   - tenant1
   - tenant2
```
So might not need a separate list command for only API.
But one thing that I would want to add is a verbose mode by which we can print out all details or all tenant and APIs.

Reply:
I think it still might be useful, especially if you have no context defined so far or if you want a higher-level view of which APIs you currently have in your config. If we can add and remove an entity, then it makes to me to list as well. But yes, it's achievable through `context` command to a degree, so it's not a blocker.

Reply:
Let's iterate on a future PR! 🙂